src/rest.cpp

```diff
@@ -570,6 +571,55 @@ static bool rest_getutxos(HTTPRequest* req, const std::string& strURIPart)
     }
 }
 
+// A bit of a hack - dependency on a function defined in rpc/mining.cpp
+UniValue estimatesmartfee(const JSONRPCRequest& request);
+
+static bool rest_getfee(HTTPRequest* req, const std::string& strURIPart) {
+    if (!CheckWarmup(req))
+        return false;
+    std::string param;
+    const RetFormat rf = ParseDataFormat(param, strURIPart);
+
+    std::vector<std::string> path;
+    boost::split(path, param, boost::is_any_of("/"));
+    path.erase(path.begin());
+
+    if (path.size() != 2)
+        return RESTERR(req, HTTP_BAD_REQUEST, "Path must be /rest/fee/<MODE>/<TARGET>.json");
+
+    // check estimation mode is valid
+    auto modestr = path[0];
+    std::transform(modestr.cbegin(), modestr.cend(), modestr.begin(), toupper);
+    FeeEstimateMode mode;
+    if (!FeeModeFromString(modestr, mode)){
+        return RESTERR(req, HTTP_BAD_REQUEST, "<MODE> must be one of <unset|economical|conservative>");
+    };
+
+
+    switch (rf) {
+    case RF_JSON: {
+
+        JSONRPCRequest jsonRequest;
+        UniValue conf_target(std::stoi(path[1]));
+        UniValue mode(modestr);
+        UniValue params(UniValue::VARR);
+        params.push_back(conf_target);
+        params.push_back(mode);
+        jsonRequest.params = params;
+
+        UniValue estimatedfee;
+        estimatedfee = estimatesmartfee(jsonRequest);
```

Comment:
Sounds weird to "redirect" to RPC handler (redo the mode parse etc). Maybe factor out what is needed from there?

Maybe just call `::feeEstimator.estimateSmartFee()` here?